File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/CSharp/main.cs
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks for a runtime error
*/

using System;

/*
 * The Program class
*/
class Program {
    static void Main() {
        // this function checks a student's grade

        // Input
        Console.Write("Enter an integer: ");
        string integerAsString = Console.ReadLine();

        // Process & Output
        try {
            int integerAsNumber = int.Parse(integerAsString);
            Console.WriteLine($"You entered: {integerAsNumber}.");
        } catch (FormatException e) {
            Console.WriteLine($"The error was: {e.Message}.");
        } finally {
            Console.WriteLine("Thanks for playing.");
        }

        Console.WriteLine("\nDone.");
    }
}

File: code_examples/3-Structured_Problem_Solving/10-Try_Catch/JavaScript/main.js
/* Created by: Mr. Coxall
 * Created on: Sep 2020
 * This program checks for a runtime error
 */

const prompt = require("prompt-sync")()

// input
const integerAsString = prompt("Enter an integer: ")

// process & output
const integerAsNumber = parseInt(integerAsString)

if (integerAsNumber != NaN) {
  console.log(`You entered: ${integerAsNumber}.`)
} else {
  console.log("Invalid input.")
}

console.log("\nDone.")
